File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/winforms/samples/controlreference/tooltipctl/vb/ChangeTooltips.vb
'-----------------------------------------------------------------------
'  This file is part of the Microsoft .NET SDK Code Samples.
' 
'  Copyright (C) Microsoft Corporation.  All rights reserved.
' 
'This source code is intended only as a supplement to Microsoft
'Development Tools and/or on-line documentation.  See these other
'materials for detailed information regarding Microsoft code samples.
' 
'THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
'KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
'IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
'PARTICULAR PURPOSE.
'-----------------------------------------------------------------------



Imports System
Imports System.ComponentModel
Imports System.Drawing
Imports System.Windows.Forms
Imports System.Diagnostics

Public Class ChangeToolTips
    Inherits System.Windows.Forms.Form
    Private toolTips() As String

    Public Sub New(ByVal smlImageList As ImageList, ByVal toolTips() As String)
        MyBase.New()
        ChangeToolTips = Me

        'This call is required by the Windows Forms Designer.
        InitializeComponent()

        Me.imgList = smlImageList
        Me.toolTips = toolTips
        listView1.SmallImageList = smlImageList
        SetToolTipRows()
    End Sub

    Private Sub SetToolTipRows()
        Debug.Assert(imgList.Images.Count = toolTips.Length, _
            "Incorrect number of images or tooltips")

        listView1.Columns.Add("ToolTip Text", listView1.Size.Width - 5, _
            HorizontalAlignment.Left)
        Dim i As Integer
        For i = 0 To toolTips.Length - 1
            Dim item As New ListViewItem(toolTips(i), i)
            listView1.Items.Add(item)
        Next i
    End Sub

    Public Overridable Function GetToolTips() As String()
        Return toolTips
    End Function

    Private Sub ListView1_afterLabelEdit(ByVal [source] As System.Object, _
        ByVal e As System.Windows.Forms.LabelEditEventArgs) _
        Handles listView1.AfterLabelEdit
        If e.Label Is Nothing Then
            e.CancelEdit = True
            Return
        End If
        toolTips(e.Item) = e.Label
    End Sub

    'Form overrides dispose to clean up the component list.
    Protected Overloads Overrides Sub Dispose(ByVal disposing as Boolean)
        If (disposing) Then
            If Not (components Is Nothing) Then
                components.Dispose()
            End If
        End If
        MyBase.Dispose(disposing)
    End Sub

    Private components As System.ComponentModel.IContainer


#Region " Windows Form Designer generated code "

    'Required by the Windows Form Designer
    Protected WithEvents btnOk As System.Windows.Forms.Button
    Protected WithEvents listView1 As System.Windows.Forms.ListView
    Protected WithEvents label1 As System.Windows.Forms.Label
    Protected WithEvents imgList As System.Windows.Forms.ImageList

    Private WithEvents ChangeToolTips As System.Windows.Forms.Form

    'NOTE: The following procedure is required by the Windows Form Designer
    'It can be modified using the Windows Form Designer.
    'Do not modify it using the code editor.
    Private Sub InitializeComponent()
        Me.components = New System.ComponentModel.Container
        Me.btnOk = New System.Windows.Forms.Button
        Me.listView1 = New System.Windows.Forms.ListView
        Me.label1 = New System.Windows.Forms.Label
        Me.imgList = New System.Windows.Forms.ImageList(Me.components)
        Me.SuspendLayout()

        '
        'btnOk
        '
        Me.btnOk.DialogResult = System.Windows.Forms.DialogResult.OK
        Me.btnOk.Location = New System.Drawing.Point(16, 272)
        Me.btnOk.Name = "btnOk"
        Me.btnOk.TabIndex = 1
        Me.btnOk.Text = "&Ok"

        '
        'listView1
        '
        Me.listView1.HeaderStyle = System.Windows.Forms.ColumnHeaderStyle.Nonclickable
        Me.listView1.LabelEdit = True
        Me.listView1.Location = New System.Drawing.Point(16, 48)
        Me.listView1.MultiSelect = False
        Me.listView1.Name = "listView1"
        Me.listView1.Size = New System.Drawing.Size(272, 192)
        Me.listView1.TabIndex = 0
        Me.listView1.Text = "listView1"
        Me.listView1.View = System.Windows.Forms.View.Details

        '
        'label1
        '
        Me.label1.Location = New System.Drawing.Point(16, 8)
        Me.label1.Name = "label1"
        Me.label1.Size = New System.Drawing.Size(272, 24)
        Me.label1.TabIndex = 3
        Me.label1.Text = "To change the text for a tooltip, click on a selected label."

        '
        'imgList
        '
        Me.imgList.ColorDepth = System.Windows.Forms.ColorDepth.Depth8Bit
        Me.imgList.ImageSize = New System.Drawing.Size(16, 16)
        Me.imgList.TransparentColor = System.Drawing.Color.Transparent

        '
        'ChangeToolTips
        '
        Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
        Me.ClientSize = New System.Drawing.Size(302, 302)
        Me.Controls.AddRange(New System.Windows.Forms.Control() {Me.label1, _
            Me.listView1, Me.btnOk})
        Me.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedDialog
        Me.MaximizeBox = False
        Me.MinimizeBox = False
        Me.Name = "ChangeToolTips"
        Me.ShowInTaskbar = False
        Me.Text = "ChangeToolTips"
        Me.ResumeLayout(False)
    End Sub
#End Region

End Class

File: vs2003_cd02/Program Files/Microsoft Visual Studio .NET 2003/SDK/v1.1/QuickStart/winforms/samples/controlreference/tooltipctl/cs/changetooltips.cs
//-----------------------------------------------------------------------
//  This file is part of the Microsoft .NET SDK Code Samples.
// 
//  Copyright (C) Microsoft Corporation.  All rights reserved.
// 
//This source code is intended only as a supplement to Microsoft
//Development Tools and/or on-line documentation.  See these other
//materials for detailed information regarding Microsoft code samples.
// 
//THIS CODE AND INFORMATION ARE PROVIDED AS IS WITHOUT WARRANTY OF ANY
//KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
//IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//PARTICULAR PURPOSE.
//-----------------------------------------------------------------------



using System;
using System.ComponentModel;
using System.Drawing;
using System.Windows.Forms;
using System.Diagnostics;


public class ChangeToolTips : System.Windows.Forms.Form {
    internal ImageList imgList;
    internal string[] toolTips;

    public ChangeToolTips(ImageList smlImageList, string[] toolTips) : base() {
        //
        // Required for Windows Form Designer support
        //
        InitializeComponent();

        this.imgList = smlImageList;
        this.toolTips = toolTips;
        listView1.SmallImageList = smlImageList;
        SetToolTipRows();
    }

    private void SetToolTipRows() {
        Debug.Assert(imgList.Images.Count == toolTips.Length,
            "Incorrect number of images or tooltips");

        listView1.Columns.Add("ToolTip Text", listView1.Size.Width - 5,
            HorizontalAlignment.Left);
        for (int i = 0; i < toolTips.Length; i++) {
            ListViewItem item = new ListViewItem(toolTips[i],i);
            listView1.Items.Add(item);
        }
    }

    public virtual string[] GetToolTips() {
        return toolTips;
    }

    /// <summary>
    ///    Clean up any resources being used.
    /// </summary>
    protected override void Dispose(bool disposing)
    {
        if (disposing) {
            if (components != null) {
                components.Dispose();
            }
        }
        base.Dispose(disposing);
    }

    private void ListView1_afterLabelEdit(object source, LabelEditEventArgs e) {
        if (e.Label == null) {
            e.CancelEdit = true;
            return;
        }
        toolTips[e.Item] = e.Label;
    }

    /// <summary>
    ///    Required designer variable.
    /// </summary>
    protected internal System.ComponentModel.Container components;
    protected internal System.Windows.Forms.Button btnOk;
    protected internal System.Windows.Forms.ListView listView1;
    protected internal System.Windows.Forms.Label label1;

    /// <summary>
    ///    Required method for Designer support - do not modify
    ///    the contents of this method with the code editor.
    /// </summary>
    private void InitializeComponent()
    {
        this.btnOk = new System.Windows.Forms.Button();
        this.listView1 = new System.Windows.Forms.ListView();
        this.label1 = new System.Windows.Forms.Label();
        this.SuspendLayout();

        // 
        // btnOk
        // 
        this.btnOk.DialogResult = System.Windows.Forms.DialogResult.OK;
        this.btnOk.Location = new System.Drawing.Point(16, 272);
        this.btnOk.Name = "btnOk";
        this.btnOk.TabIndex = 1;
        this.btnOk.Text = "&Ok";

        // 
        // listView1
        // 
        this.listView1.HeaderStyle = System.Windows.Forms.ColumnHeaderStyle.Nonclickable;
        this.listView1.LabelEdit = true;
        this.listView1.Location = new System.Drawing.Point(16, 48);
        this.listView1.MultiSelect = false;
        this.listView1.Name = "listView1";
        this.listView1.Size = new System.Drawing.Size(272, 192);
        this.listView1.TabIndex = 0;
        this.listView1.Text = "listView1";
        this.listView1.View = System.Windows.Forms.View.Details;
        this.listView1.AfterLabelEdit += 
            new System.Windows.Forms.LabelEditEventHandler(this.ListView1_afterLabelEdit);

        // 
        // label1
        // 
        this.label1.Location = new System.Drawing.Point(16, 8);
        this.label1.Name = "label1";
        this.label1.Size = new System.Drawing.Size(272, 24);
        this.label1.TabIndex = 3;
        this.label1.Text = "To change the text for a tooltip, click on a selected label.";

        // 
        // ChangeToolTips
        // 
        this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
        this.ClientSize = new System.Drawing.Size(302, 302);
        this.Controls.AddRange(new System.Windows.Forms.Control[] {
            this.label1, this.listView1,this.btnOk});
        this.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedDialog;
        this.MaximizeBox = false;
        this.MinimizeBox = false;
        this.Name = "ChangeToolTips";
        this.ShowInTaskbar = false;
        this.Text = "ChangeToolTips";
        this.ResumeLayout(false);
    }
}
